### ISSUE TITLE
[Android][OpenGL&Vulkan][Not enough FPS] The KING of Fighters Portable '94-'98 Chapter of Orochi (NPJH50258)

### ISSUE THREAD

**Issue by nassau-tk** [hrydgard/ppsspp#12543]
### What happens?
Not enough(stable) FPS.

### What *should* happen?
Work on stable FPS.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

PPSSPP Ver. : 1.9.3-164
OS: Android 9.0
DeviceName: SONY SO-02K / Xperia XZ1 Compact (Japan)
CPU: Snapdragon835 (Octa-core)
GPU: Adreno 540
Resolution: 720×1280
RAM: 4GB
ROM:32GB
ExternalStrage: microSDXC 512GB

I attached dump.
Please,check it.

[NPJH50258_0001.zip](https://github.com/hrydgard/ppsspp/files/4014792/NPJH50258_0001.zip)

**Comment by Emulatorer** [comment]
Similar issue  #11035 hope sir @unknownbrackets or sir @hrydgard can fixed this issue 👍

**Comment by unknownbrackets** [comment]
That dump looks like an intro screen.  Is it already slow even at that point?  It must not be graphics related, if so.

-[Unknown]

**Comment by nassau-tk** [comment]
Really?!
This lssue does not happen intro and any menu screen.
This issue happen on the playing screen.

I  tried getting the dump.
Please check it.

[NPJH50258_0002.zip](https://github.com/hrydgard/ppsspp/files/4275715/NPJH50258_0002.zip)

**Comment by Gamemulatorer** [comment]
No improvement on the latest version of ppsspp :(
![Screenshot_2021-02-15-02-31-11](https://user-images.githubusercontent.com/37603562/107885700-cd681680-6f36-11eb-9ca0-255e39ce626a.png)
![Screenshot_2021-02-15-02-38-23](https://user-images.githubusercontent.com/37603562/107885718-e670c780-6f36-11eb-9f64-c9d13fa3a9ca.png)

**Comment by Panderner** [comment]
This game runs full speed for me in all games.
![Screenshot_2021-02-15-18-57-21-647_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/107938114-d3eea080-6fbf-11eb-8a30-d1becfd6a3ed.jpg)
But the problem is this game requires the rendering mode to be set to buffered.

**Comment by Gamemulatorer** [comment]
> This game runs full speed for me in all games.
Because you are not using the Japanese version lol

**Comment by Panderner** [comment]
Oops I mean japanese version of course can't run full speed and sprite glitches.

https://user-images.githubusercontent.com/37503397/107942210-e53aab80-6fc5-11eb-9e6b-7290fdf77636.mp4

**Comment by Gamemulatorer** [comment]
I added this game to force software rendering it's better compare to hardware but still not smooth as the USA version.
I also encounter some graphics issue when this game is in software rendering.
![Screenshot_2021-08-23-16-33-45-867_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/130416900-b6ef34c8-6633-4807-8c04-64067439b5ae.jpg)

Minimizing ppsspp can cause this both Issue
![Screenshot_2021-08-23-16-34-02-383_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/130416933-232a74a6-ff63-48f0-a48f-2bc90686f39e.jpg)
Some of the touchscreen buttons is missing
![Screenshot_2021-08-23-16-34-10-369_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/130417009-65a4c5a5-32e4-40db-bd78-da49c47edde0.jpg)

**Comment by Gamemulatorer** [comment]
![Screenshot_2021-08-23-22-41-07-195_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/130467147-e12636fb-ec11-4a60-83cc-ebb3026bc5c2.jpg)
[NPJH50258.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/7032782/NPJH50258.ppdmp.zip)

**Comment by Gamemulatorer** [comment]
This game is much faster/smoother on software rendering compare to hardware rendering.

HW Vulkan 1X Resolution

https://user-images.githubusercontent.com/37603562/150362858-cd1fed21-cf99-49af-aba1-e2189858816a.mp4

**Software Mode**

https://user-images.githubusercontent.com/37603562/150363077-8c90ed0c-7e3f-4f51-8d72-6d42ae4dc94f.mp4

Add this game to ForceSoftwareRendering compat.ini
# KOF Chapter of Orochi
NPJH50258 = true

**Comment by hrydgard** [comment]
Yeah that might be reasonable to do for this game.

**Comment by unknownbrackets** [comment]
It might be time to bring back the software renderer option on Android?

-[Unknown]

**Comment by Panderner** [comment]
There is also side effect for this game in software mode that causes large performance slow down in menus.
![Screenshot (17)](https://user-images.githubusercontent.com/37503397/150480488-270227ff-5515-42eb-a50d-a5d6d38116e9.png)

**Comment by Gamemulatorer** [comment]
> It might be time to bring back the software renderer option on Android?
> 
> -[Unknown]

Yes!

**Comment by Gamemulatorer** [comment]
There's a bug both for software and hardware.
The background stage of Orochi Iori should turn to red color but its became black and white.
![Screenshot_2022-01-23-11-27-36-861_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/150663806-d68b10e1-e3ee-4756-830e-fb45eeedfea8.jpg)
[recording.ppdmp.zip](https://github.com/hrydgard/ppsspp/files/7919686/recording.ppdmp.zip)

**Comment by unknownbrackets** [comment]
If that's incorrect, it would have to be something on the CPU.  Sounds like that other fighting game, where the effects are gray...

I didn't see any palettes that looked promising for a red background, so it seems like a CLUT issue maybe even?

PSP rendering of that dump:
![#12543_NPJH50258_king_of_fighters_gray](https://user-images.githubusercontent.com/191233/150707003-dacf7365-b298-4dd5-9812-920ebd32bf78.png)

-[Unknown]

**Comment by Gamemulatorer** [comment]
Maybe that's a game bug? and not by ppsspp?
This is the correct color of the Orochi Iori stage.
![Screenshot_2022-01-24-10-21-28-598_com vanced android youtube](https://user-images.githubusercontent.com/37603562/150711985-f09ee0e3-2355-40d2-ba77-6e68e4923fd7.jpg)

**Comment by hrydgard** [comment]
Yeah missing colorization of a palette sound suspiciously like Soul Calibur, which we still have no idea what's going on with the sword swinging effects in ...

**Comment by hrydgard** [comment]
Since the software renderer option is now available on Android, I'm removing this from the 1.13 milestone.

We could force it on for the game in the compat.ini file but then some menus may be slow as reported by Panderner, not sure if that's acceptable.

**Comment by Unknowncat1991** [comment]
This game is good to close now?
Redmi 15 5G snapdragon 6s gen3

<img width="2340" height="1080" alt="Image" src="https://github.com/user-attachments/assets/caa16e73-687f-4e15-9944-fb9b6a9e1e32" />

<img width="2340" height="1080" alt="Image" src="https://github.com/user-attachments/assets/3e9029a5-58f8-40d3-a05a-2375b44c8fe5" />

**Comment by nassau-tk** [comment]
I believe there were a few issues with this game, but I think it is best to leave the decision of whether to close this to @hrydgard.
